File: sol/solution/0001-0100/0038. Count and Say/Solution.ts
function countAndSay(n: number): string {
    let res = "1";
    for (let i = 2; i <= n; i++) {
        let cur = "";
        let cnt = 1;
        for (let j = 1; j < res.length; j++) {
            if (res[j] === res[j-1]) cnt++;
            else {
                cur += cnt.toString() + res[j-1];
                cnt = 1;
            }
        }
        cur += cnt.toString() + res[res.length-1];
        res = cur;
    }
    return res;
}

File: sol/solution/0001-0100/0038. Count and Say/Solution.rs
impl Solution {
    pub fn count_and_say(n: i32) -> String {
        let mut res = "1".to_string();
        for _ in 2..=n {
            let mut cur = String::new();
            let mut cnt = 1;
            let chars: Vec<char> = res.chars().collect();
            for j in 1..chars.len() {
                if chars[j] == chars[j-1] {
                    cnt += 1;
                } else {
                    cur.push_str(&cnt.to_string());
                    cur.push(chars[j-1]);
                    cnt = 1;
                }
            }
            cur.push_str(&cnt.to_string());
            cur.push(*chars.last().unwrap());
            res = cur;
        }
        res
    }
}